src/api_proxy/service_control/request_builder_test.cc

```diff
@@ -305,10 +306,63 @@ TEST_F(RequestBuilderTest, FillReportRequestFailedTest) {
   ASSERT_EQ(expected_text, text);
 }
 
+TEST_F(RequestBuilderTest, FillReportWithInvalidApiKeyTest) {
+  ReportRequestInfo info;
+  FillOperationInfo(&info);
+  FillReportRequestInfo(&info);
+
+  // Test case where API key is invalid.
+  // The key point is the API Key will be included in the log entry,
+  // but not the `credential_id` or `consumer_id` metrics.
+  info.api_key = "invalid-api-key";
+  info.check_response_info.api_key_state = api_key::ApiKeyState::INVALID;
+
+  // Use 401 as a failed response code.
+  info.response_code = 401;
+
+  // Use the corresponding status for that response code.
+  info.status = Status(Code::PERMISSION_DENIED, "");
+
+  gasv1::ReportRequest request;
+  ASSERT_TRUE(scp_.FillReportRequest(info, &request).ok());
+
+  std::string text = ReportRequestToString(&request);
+  std::string expected_text =
+      ReadTestBaseline("report_request_invalid_api_consumer.golden");
+  ASSERT_EQ(expected_text, text);
+}
+
+TEST_F(RequestBuilderTest, FillReportWithNotEnabledApiKeyTest) {
```

Comment:
Maybe combine these two tests into one, with a loop to test all 3 states:  INVALID, NOT_ENABLED, NOT_CHECKED.

Reply:
Done